pymc_marketing/clv/distributions.py

```diff
@@ -476,3 +475,207 @@ def logp(value, r, alpha, s, beta, T):
+                    recency if x > 0 else T,
+                    x,
+                ],
+            )
+
+        for index in np.ndindex(*size):
+            output[index] = sim_data(purchase_prob[index], churn_prob[index], T[index])
+
+        return output
+
+    def _supp_shape_from_params(*args, **kwargs):
+        return (2,)
+
+
+beta_geo_beta_binom = BetaGeoBetaBinomRV()
+
+
+class BetaGeoBetaBinom(Discrete):
+    r"""
+    Population-level distribution class for a discrete, non-contractual, Beta-Geometric/Beta-Binomial process,
+    based on Fader, et al. in [1]_.
+
+    .. math::
+
+        \mathbb{L}(\alpha, \beta, \gamma, \delta  | x, t_x, n) &=
+        \frac{B(\alpha+x,\beta+n-x)}{B(\alpha,\beta)}
+        \frac{B(\gamma,\delta+n)}{B(\gamma,\delta)} \\
+        &+ \sum_{i=0}^{n-t_x-1}\frac{B(\alpha+x,\beta+t_x-x+i)}{B(\alpha,\beta)} \\
+        &\cdot \frac{B(\gamma+1,\delta+t_x+i)}{B(\gamma,\delta)}
+
+    ========  ===============================================
+    Support   :math:`t_j >= 0` for :math:`j = 1, \dots, x`
+    Mean      :math: `\mathbb{E}[X(n) | \alpha, \beta, \gamma, \delta] =  (\frac{\alpha}{\alpha+\beta})(\frac{\delta}{\gamma-1}) \cdot{1-\frac{\Gamma(\gamma+delta}{\Gamma(\gamma+\delta+n}\frac{\Gamma(1+\delta+n}{\Gamma(1+ \delta}}`
+    ========  ===============================================
+
+    References
+    ----------
+    .. [1] Fader, Peter S., Bruce G.S. Hardie, and Jen Shang (2010),
+       "Customer-Base Analysis in a Discrete-Time Noncontractual Setting,"
+       Marketing Science, 29 (6), 1086-1108. https://www.brucehardie.com/papers/020/fader_et_al_mksc_10.pdf
+
+    """
+
+    rv_op = beta_geo_beta_binom
+
+    @classmethod
+    def dist(cls, alpha, beta, gamma, delta, T, **kwargs):
+        return super().dist([alpha, beta, gamma, delta, T], **kwargs)
+
+    def logp(value, alpha, beta, gamma, delta, T):
+        t_x = pt.atleast_1d(value[..., 0])
+        x = pt.atleast_1d(value[..., 1])
+        scalar_case = t_x.type.broadcastable == (True,)
+
+        for param in (t_x, x, alpha, beta, gamma, delta, T):
+            if param.type.ndim > 1:
+                raise NotImplementedError(
+                    f"BetaGeoBetaBinom logp only implemented for vector parameters, got ndim={param.type.ndim}"
+                )
+            if scalar_case:
+                if param.type.broadcastable == (False,):
+                    raise NotImplementedError(
+                        f"Parameter {param} cannot be larger than scalar value"
+                    )
+
+        # Broadcast all the parameters so they are sequences.
+        # Potentially inefficient, but otherwise ugly logic needed to unpack arguments in the scan function,
+        # since sequences always precede non-sequences.
+        _, alpha, beta, gamma, delta, T = pt.broadcast_arrays(
+            t_x, alpha, beta, gamma, delta, T
+        )
+
+        def logp_customer_died(t_x_i, x_i, alpha_i, beta_i, gamma_i, delta_i, T_i):
```

Comment:
Can we add a reference to the equation this formal uses in https://www.brucehardie.com/papers/020/fader_et_al_mksc_10.pdf?  (is it the log of equation (5) ?)